selenium/che-selenium-core/src/main/java/org/eclipse/che/selenium/core/client/TestGitHubRepository.java

```diff
@@ -174,4 +153,60 @@ public String getHtmlUrl() {
   public String getSshUrl() {
     return ghRepo.getSshUrl();
   }
+
+  private GHRepository create() throws IOException, InterruptedException {
+    GHRepository repo = gitHub.createRepository(repoName).create();
+    ensureRepositoryCreated(repo, System.currentTimeMillis());
+
+    LOG.info("GitHub repo {} has been created", repo.getHtmlUrl());
+    return repo;
+  }
+
+  private void ensureRepositoryCreated(GHRepository repo, long startCreationTimeInMillisec)
+      throws IOException {
+    Throwable lastIOException = null;
+    for (int i = 0; i < REPO_CREATION_ATTEMPTS; i++) {
+      try {
+        gitHub.getRepository(repo.getFullName());
+        return;
+      } catch (IOException e) {
+        lastIOException = e;
+        LOG.info("Waiting for {} to be created", repo.getHtmlUrl());
+        sleepQuietly(GITHUB_OPERATION_TIMEOUT_SEC); // sleep one second
+      }
+    }
+
+    long durationOfRepoCreationInSec =
+        (System.currentTimeMillis() - startCreationTimeInMillisec) / 1000;
+
+    throw new IOException(
+        format(
+            "GitHub repo %s hasn't been created in %s seconds",
+            repo.getHtmlUrl(), durationOfRepoCreationInSec),
+        lastIOException);
+  }
+
+  /**
+   * Creates file in GitHub repository.
+   *
+   * @param pathToRootContentDirectory path to the root directory of file locally
+   * @param pathToFile path to file locally
+   * @throws IOException
+   */
+  private void createFile(Path pathToRootContentDirectory, Path pathToFile) throws IOException {
+    byte[] contentBytes = Files.readAllBytes(pathToFile);
+    String relativePath = pathToRootContentDirectory.relativize(pathToFile).toString();
+    String commitMessage = String.format("Add file %s", relativePath);
```

Comment:
String commitMessage = "Add file " + relativePath;

Reply:
I would prefer String.format()

Reply:
But in my opinion in this case the concatenation is better for use. But of course, this is only my opinion.